KDI Importer/csv_KDI_json.rb

```diff
@@ -18,341 +20,340 @@
-        asset = $assets.select{|a| a[:external_id] == external_id }.first
-      when "database"
-        asset = $assets.select{|a| a[:database] == database }.first
-      else
-        "Error: main locator not provided" if @debug
-    end
+      tmpassets << { file: file.to_s } unless file.nil? || file.empty?
+      tmpassets << { ip_address: ip_address } unless ip_address.nil? || ip_address.empty?
+      tmpassets << { mac_address: mac_address } unless mac_address.nil? || mac_address.empty?
+      tmpassets << { hostname: hostname } unless hostname.nil? || hostname.empty?
+      tmpassets << { ec2: ec2.to_s } unless ec2.nil? || ec2.empty?
+      tmpassets << { netbios: netbios.to_s } unless netbios.nil? || netbios.empty?
+      tmpassets << { url: url.to_s } unless url.nil? || url.empty?
+      tmpassets << { fqdn: fqdn.to_s } unless fqdn.nil? || fqdn.empty?
+      tmpassets << { external_id: external_id.to_s } unless external_id.nil? || external_id.empty?
+      tmpassets << { database: database.to_s } unless database.nil? || database.empty?
+      tmpassets << { application: application.to_s } unless application.nil? || application.empty?
+      tmpassets << { tags: tags } unless tags.nil? || tags.empty?
+      tmpassets << { owner: owner.to_s } unless owner.nil? || owner.empty?
+      tmpassets << { os: os.to_s } unless os.nil? || os.empty?
+      tmpassets << { os_version: os_version.to_s } unless os_version.nil? || os_version.to_s.empty?
+      tmpassets << { priority: priority } unless priority.nil? || priority.to_s.empty?
+      tmpassets << { vulns: [] }
+
+      if file.to_s.empty? && ip_address.to_s.empty? && mac_address.to_s.empty? && hostname.to_s.empty? && ec2.to_s.empty? && netbios.to_s.empty? && url.to_s.empty? && database.to_s.empty? && external_id.to_s.empty? && fqdn.to_s.empty? && application.to_s.empty?
+        success = false
+      end
 
-    puts "Unknown asset, can't associate a vuln!" unless asset
-    return unless asset
+      $assets << tmpassets.reduce(&:merge) if success
 
-    # associate the asset
-    assetvulns = []
-    assetvulns << {:scanner_type => "#{scanner_type}",:scanner_identifier => "#{scanner_id}",}
-    assetvulns << {:details => "#{details}"} unless details.nil?
-    assetvulns << {:created_at => "#{created}"} unless created.nil?
-    assetvulns << {:scanner_score => scanner_score} unless scanner_score.nil? || scanner_score == 0
-    assetvulns << {:last_fixed_on => "#{last_fixed}"} unless last_fixed.nil?
-    assetvulns << {:last_seen_at => "#{last_seen}"} unless last_seen.nil?
-    assetvulns << {:closed_at => "#{closed}"} unless closed.nil?
-    assetvulns << {:port => port,} unless port.nil?
-    assetvulns << {:status => "#{status}"}
+      success
+    end
 
-    asset[:vulns] << assetvulns.reduce(&:merge)
+    def create_asset_vuln(hostname, ip_address, file, mac_address, netbios, url, ec2, fqdn, external_id, database, scanner_type, scanner_id, details, created, scanner_score, last_fixed,
```

Comment:
not formatting related so pretty irrelevant, but this is waaaayyy too many arguments. for something like this its sometimes good to make a single object that contains all of these params as properties. or maybe break down the method in a different way.